### PR TITLE
Add a non default service account

### PR DESCRIPTION
**Pull request by kkavitha**
Pull request

### What this PR does / why we need it
https://jira.eng.vmware.com/browse/TAPI-76 - Adding an overlay to add annotation to service account with arn role id for AWS

### CODE REVIEW

File: config/samples/v1alpha1_podintent.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: PodIntent
 metadata:
   name: podintent-sample
 spec:
-  serviceAccountName: default
+  serviceAccountName: controller-manager
```
[the remark this answers is not in the document]
Comment (kkavitha) [reply config/samples/v1alpha1_podintent.yaml:6]: So, the service account on the pod intent is not required to be the same as the one the convention controller uses?

[the remark this answers is not in the document]
Comment (kkavitha) [reply config/samples/v1alpha1_podintent.yaml:6]: Oh no.. I got it. The namespace can be anything. not the `cartographer-system` namespace!

File: go.work.sum [hunk 2]
```diff
@@ -1,17 +1,406 @@
+4d63.com/gochecknoglobals v0.1.0/go.mod h1:wfdC5ZjKSPr7CybKEcgJhUOgeAQW1+7WcyK8OvUilfo=
```
Comment (scothis) [inline go.work.sum:1]: I wouldn't expect any of the go dependencies to change as part of this PR

**Review by scothis — APPROVED** [review]

File: config/rbac/service-account.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
```
[the remark this answers is not in the document]
Comment (kkavitha) [reply config/rbac/service-account.yaml:1]: yeah, I was going back and forth but the rbac layer had the maximum number of references 😄  I can move it to manager folder

File: go.work.sum (hunk 2 above)
Comment (kkavitha) [reply go.work.sum:1]: yeah, running `make dist` changes it for me. This happens even on main with the latest commit. I have go 1.18.1 on my system.

**Pull request merged**